pkg/operation/botanist/component/etcdcopybackupstask/etcdcopybackupstask_test.go

```diff
@@ -0,0 +1,248 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package etcdcopybackupstask_test
+
+import (
+	"context"
+	"time"
+
+	"github.com/gardener/gardener/pkg/logger"
+	mockclient "github.com/gardener/gardener/pkg/mock/controller-runtime/client"
+	. "github.com/gardener/gardener/pkg/operation/botanist/component/etcdcopybackupstask"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+var _ = Describe("EtcdCopyBackupsTask", func() {
+	var (
+		ctrl *gomock.Controller
+		ctx  context.Context
+		c    *mockclient.MockClient
```

Comment:
General note: We should strive for not using mocks in such tests anymore since they are hard to maintain. For newly added components we started to work with fake clients.
Feel free to ignore this for now if you want, but I wanted to point it out for the future.

Reply:
@rfranzke I don't fully understand (and in general disagree) with the "mocks are hard to maintain" statement. I would propose to have a discussion about this (and also in the future in similar situations) before we adopt fake clients as replacement for mocks.
I believe the issues / objections you may have with mocks have more to do with the way mocks are being used, rather than with any inherent "hardness to maintain".

Reply:
Yes, that's for sure also a point on the agenda in this test strategy meeting in CW48. I don't suggest to replace mocks in general, but I think for the component use-case they are indeed hard to maintain and provide little value (or rather no?) compared to using fake clients. In my experience (after writing several components with both tests based on mock clients and tests based on fake clients), I clearly tend towards the latter.
Maybe we use them in a wrong way, but I think https://github.com/gardener/gardener/pull/4227 is also a good example of why they are "hard to maintain". In this PR, we switched from `Update` to `Patch` and had to adopt basically all components. IMO this is an implementation detail and should not matter to the unit test of the component. If we had used a fake client here then no such changes would have been necessary.

Reply:
OK, this is certainly an interesting discussion but I would suggest not to rework this PR at this point, unless you have strong objections.